app/models/project.rb

```diff
@@ -271,7 +271,15 @@ def homepage
   def set_dependents_count
     return if destroyed?
     new_dependents_count = dependents.joins(:version).pluck('DISTINCT versions.project_id').count
-    new_dependent_repos_count = dependent_repositories.open_source.count.length
+
+    # Most of the time we truncate this count if it is larger than 1,000 anyways, so only do slow version if less
```

Comment:
Is the reason to do this just to avoid stale data? It's probably fine to just accept the eventual consistency and always go to the view